library/std/src/net/ip.rs

```diff
@@ -319,15 +319,7 @@ impl Ipv4Addr {
     #[stable(feature = "rust1", since = "1.0.0")]
     #[rustc_const_stable(feature = "const_ipv4", since = "1.32.0")]
     pub const fn new(a: u8, b: u8, c: u8, d: u8) -> Ipv4Addr {
-        // FIXME: should just be u32::from_be_bytes([a, b, c, d]),
-        // once that method is no longer rustc_const_unstable
-        Ipv4Addr {
-            inner: c::in_addr {
-                s_addr: u32::to_be(
-                    ((a as u32) << 24) | ((b as u32) << 16) | ((c as u32) << 8) | (d as u32),
-                ),
-            },
-        }
+        Ipv4Addr { inner: c::in_addr { s_addr: u32::from_be_bytes([a, b, c, d]) } }
```

Comment:
maybe the fixme was wrong? I can never get be/le order right...

Reply:
Jup, it's wrong: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2920668bc321a75501f2bc033ed9f893

I guess we need `from_le_bytes` here. Makes sense since the function says the bytes already have to be in the right order.

Reply:
Nice. Will run the test locally first and update the PR later.

Reply:
Think about it again. Do we guarantee that `c::in_addr` in big endian?
I get the feelings that we are missing something.

Reply:
I would expect `u32::from_ne_bytes([a, b, c, d])`, since we want to store those as `abcd` (an internet address is stored in big-endian byte ordering). This would be equivalent to the old implementation. The `u32::from_le_bytes([a, b, c, d])` swaps the order on big-endian targets to `dcba`.

Reply:
Those endiannesses is confusing in practice. Hope we have an way to enforce this in type system.

Reply:
Yeah, judge by assembly, I think ne is the correct answer: https://rust.godbolt.org/z/e5ahnK

Reply:
Very likely that I was wrong in the TODO -- I think I was wrong in https://github.com/rust-lang/rust/pull/57740#discussion_r249326435 too